cosi/cosi.go

```diff
@@ -328,123 +351,162 @@ func (c *CoSi) genResponse() error {
+func (cm *mask) SetMaskBit(signer int, enable bool) {
 	if signer > len(cm.publics) {
-		panic("SetMaskBit range out of index")
+		panic("SetMaskBit index out of range")
 	}
 	byt := signer >> 3
-	bit := byte(1) << uint(signer&7)
-	if !enabled {
-		if cm.mask[byt]&bit == 0 { // was enabled
-			cm.mask[byt] |= bit // disable it
-			cm.aggPublic.Sub(cm.aggPublic, cm.publics[signer])
-		}
-	} else { // enable
-		if cm.mask[byt]&bit != 0 { // was disabled
-			cm.mask[byt] &^= bit
-			cm.aggPublic.Add(cm.aggPublic, cm.publics[signer])
-		}
+	msk := byte(1) << uint(signer&7)
+	if ((cm.mask[byt] & msk) == 0) && enable {
+		cm.mask[byt] ^= msk // flip bit in mask from 0 to 1
+		cm.aggPublic.Add(cm.aggPublic, cm.publics[signer])
+	}
+	if ((cm.mask[byt] & msk) != 0) && !enable {
+		cm.mask[byt] ^= msk // flip bit in mask from 1 to 0
+		cm.aggPublic.Sub(cm.aggPublic, cm.publics[signer])
 	}
 }
 
-// MaskBit returns a boolean value indicating whether
-// the indicated signer is enabled (true) or disabled (false)
+// MaskBit returns a boolean value indicating whether the given signer is
+// enabled (true) or disabled (false).
 func (cm *mask) MaskBit(signer int) bool {
 	if signer > len(cm.publics) {
-		panic("MaskBit given index out of range")
+		panic("MaskBit index out of range")
 	}
 	byt := signer >> 3
-	bit := byte(1) << uint(signer&7)
-	return (cm.mask[byt] & bit) != 0
+	msk := byte(1) << uint(signer&7)
+	return (cm.mask[byt] & msk) != 0
 }
 
-// bytes returns the byte representation of the mask
-// The bits that are left are set to a default value (1) for
-// non malleability.
-func (cm *mask) bytes() []byte {
-	clone := make([]byte, len(cm.mask))
-	for i := range clone {
-		clone[i] = 0xff
+// CountEnabled returns the number of enabled nodes in the CoSi participation
+// mask, i.e., it returns the hamming weight of the mask.
+func (cm *mask) CountEnabled() int {
+	hw := 0
+	for i := range cm.publics {
+		if cm.MaskBit(i) {
+			hw++
+		}
 	}
-	copy(clone[:], cm.mask)
-	return clone
+	return hw
+}
+
+// CountTotal returns the total number of nodes this CoSi instance knows.
+func (cm *mask) CountTotal() int {
+	return len(cm.publics)
 }
 
-// Aggregate returns the aggregate public key of all *participating* signers
+// Aggregate returns the aggregate public key of all *participating* signers.
 func (cm *mask) Aggregate() abstract.Point {
 	return cm.aggPublic
 }
+
+// Policy represents a fully customizable cosigning policy deciding what
+// cosigner sets are and aren't sufficient for a collective signature to be
+// considered acceptable to a verifier. The Check method may inspect the set of
+// participants that cosigned by invoking cosi.Mask and/or cosi.MaskBit, and may
+// use any other relevant contextual information (e.g., how security-critical
+// the operation relying on the collective signature is) in determining whether
+// the collective signature was produced by an acceptable set of cosigners.
+type Policy interface {
+	Check(c *CoSi) bool
```

Comment:
Good point. The participants (i.e., their public keys) are currently stored inside the `mask`, so passing the `mask` alone should be sufficient.